packages/utils/src/bulk-update-patients.ts

```diff
@@ -0,0 +1,167 @@
+ * 1. Set the env vars:
+ *  -CX_ID
+ *  -API_KEY
+ *  -API_URL
+ *  -DB_CREDS - Must use the read replica
+ * 2. Set the date string in the format YYYY-MM-DD
+ * 3. Run the script with `ts-node src/bulk-update-patients.ts`
+ */
+async function main() {
+  const dbCreds = JSON.parse(sqlDBCreds);
+  if (!dateString) {
+    console.log("Please provide a date string in the format YYYY-MM-DD");
+    return;
+  }
+  const targetDate = new Date(dateString);
+
+  const sequelize = new Sequelize(dbCreds.dbname, dbCreds.username, dbCreds.password, {
+    host: dbCreds.host,
+    port: dbCreds.port,
+    dialect: dbCreds.engine,
+  });
+
+  try {
+    const facilities = await metriportAPI.listFacilities();
+    const patientCQLinks = await getPatientCqLinks(sequelize, cxId);
+
+    let totalFacilities = 0;
+    let totalPatients = 0;
+    let patientsNoLinks = 0;
+
+    for (let i = 0; i < facilities.length; i++) {
+      const facility = facilities[i];
+      totalFacilities++;
+      const patientsList = await metriportAPI.listPatients(facility.id);
+
+      const patientsCreatedAfter = patientsList.filter(
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        patient => new Date(patient.dateCreated!) > targetDate
+      );
+
+      totalPatients += patientsCreatedAfter.length;
+
+      const patientsWithNoLinks = [];
+
+      for (const patient of patientsCreatedAfter) {
+        const hasPatientLinks = patientCQLinks.filter(link => link.id === patient.id);
+        if (hasPatientLinks.length > 0) {
+          continue;
+        }
+
+        patientsWithNoLinks.push(patient);
+        patientsNoLinks++;
+      }
+
+      const patientChunks = chunk(patientsWithNoLinks, PATIENT_CHUNK_SIZE);
+      console.log(`Facility ${facility.id} has ${patientsWithNoLinks.length} patients`);
+
+      for (const [i, patients] of patientChunks.entries()) {
+        console.log(`Chunk ${i + 1} of ${patientChunks.length}`);
+        console.log(`# of patients ${patients.length}`);
+
+        for (const patient of patients) {
+          console.log(`Updating patient ${patient.id}`);
+          const addressObject = Array.isArray(patient.address)
+            ? patient.address[0]
+            : patient.address;
+          const address = addressObject as Address;
+          await updatePatient(patient, address, facility.id);
+        }
+        if (i < patientChunks.length - 1) {
+          await sleep(CHUNK_DELAY_MAX_MS * patients.length);
+        }
+      }
+      if (i < facilities.length - 1) {
+        await sleep(CHUNK_DELAY_MAX_MS);
+      }
+    }
+
+    console.log(`Total facilities: ${totalFacilities}`);
+    console.log(`Total patients: ${totalPatients}`);
+    console.log(`Patients with no links: ${patientsNoLinks}`);
+  } catch (err) {
+    console.error(err);
+  } finally {
+    sequelize.close();
+  }
+}
+
+main();
+
+async function getPatientCqLinks(sequelize: Sequelize, cxId: string): Promise<CqPatientDataResult> {
```

Comment:
nit: put both helper functions at top of file

Reply:
There's no definition for this, I personally prefer the main function at the top, that's how we read code, from the main logic to the detail anyways.

https://www.notion.so/metriport/Development-Best-Practices-bb3574d3c83d46dc9914a9e5c5542ca7?pvs=4#de6ccc0884ec4c8c976858ec476b34be